service/controller/v5/resource_set.go

```diff
@@ -166,6 +167,26 @@ func NewResourceSet(config ResourceSetConfig) (*controller.ResourceSet, error) {
 		}
 	}
 
+	var containerURLResource controller.Resource
+	{
+		azureClients, err := client.NewAzureClientSet(config.HostAzureClientSetConfig)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+		storageAccountsClient := azureClients.StorageAccountsClient
```

Comment:
Are you sure this is the right way to configure the clients? I remember that is different with BYOC. Can you double check? I feel like I miss something here.

Reply:
I see how whole `HostAzureClientsetConfig` is passed into resources. But I need only StorageAccountsClient. And it is implemented in the same as all other clients, so I don't see how that can be different even with byoc case. Maybe @kopiczko or @TheoBrigitte can suggest something if this approach is wrong.

Reply:
The question is if this should really be the client talking to control plane subscription. Shouldn't it be a client talking to  tenant subscription?